src/annotator/util/buckets.js

```diff
@@ -0,0 +1,248 @@
+import { getBoundingClientRect } from '../highlighter';
+
+/**
+ * @typedef {import('../../types/annotator').Anchor} Anchor
+ */
+
+/**
+ * A structured Array representing either the top (`startOrEnd` = 1) or the
+ * bottom (`startOrEnd` = -1) of an anchor's highlight bounding box.
+ *
+ * @typedef {[pixelPosition: number, startOrEnd: (-1 | 1), anchor: Anchor]} PositionPoint
+ */
+
+/**
+ * An object containing information about anchor highlight positions
+ *
+ * @typedef PositionPoints
+ * @prop {Anchor[]} above - Anchors that are offscreen above
+ * @prop {Anchor[]} below - Anchors that are offscreen below
+ * @prop {PositionPoint[]} points - PositionPoints for on-screen anchor
+ *   highlights. Each highlight box has 2 PositionPoints (one for top edge
+ *   and one for bottom edge).
+ */
+
+/**
+ * @typedef BucketInfo
+ * @prop {Array<Anchor[]>} buckets
+ * @prop {number[]} index - Array of (pixel) positions of each bucket
```

Comment:
I plan to rename this property (and/or refactor the structure of this data) as soon as is feasible.

Reply:
Makes sense. This might also address the comment I had in `bucket-bar.js` about unifying the `buckets` and `index` arrays.